### PR TITLE
SEIR agentbased model

### PR DESCRIPTION
**Pull request by kuncao0228**
modified agentbased2.jl to include E state for SEIR model, updated comments and documentation to reflect SEIR model

### CODE REVIEW

File: examples/agentbasedSEIR.jl [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+# 4. $R$ Recovered
+#
+# The agents go from `S->E`, `E->I`, `I-R`, and `R->S` based on random numbers. The probability of S-> is dependent on the fraction of agents in state :E. The probability of recovering is a constant ρ, the probability of exposure is denoted by variable β, and the disease confers some temporary immunity with probability μ.
+#
+# This script has an entrypoint to call it so that you can include this file and run as many simulations as you want. The intended use case is to repeatedly call `main` and accumulate the return values into an array for later analysis.
+
+ρ = 0.5 + randn(Float64)/4 # chance of recovery
+μ = 0.5 # chance of immunity
+β = 0.5 # chance of exposure
+function transition(sm::StateModel, i::Int, s::Symbol)
```
Comment (jpfairbanks) [inline examples/agentbasedSEIR.jl]: So this has one function with a bunch of switching logic. The other version has one function per state and uses a dictionary to implement this switching logic. Which is better?

Comment (kuncao0228) [reply examples/agentbasedSEIR.jl]: Perhaps in terms of cleanliness the dictionary implementation would be better. But may be harder to demonstrate in an automated flow in the demo. But I can do both, if you would like this in agentbased.jl?

Comment (cl4yton) [reply examples/agentbasedSEIR.jl]: A suggestion: function per state is something we can handle immediately; we are still working on structured types. We're just finishing work on arrays, dictionaries (hash) are next.

Comment (jpfairbanks) [reply examples/agentbasedSEIR.jl]: Ok that is great feedback @cl4ton. We should go with the function per state approach.

**Pull request closed without merging**